drivers/include/isl29125.h

```diff
@@ -115,13 +146,23 @@ typedef struct {
  * @param[in] mode          operation mode
  * @param[in] range         measurement range
  * @param[in] resolution    AD conversion resolution
+ * @param[in] interrupt_status          Interrupt status
+ * @param[in] interrupt_persist         Interrupt persistency
+ * @param[in] interrupt_conven          RGB conversion done to interrupt control, enable
+ * @param[in] lower_threshold           Lower interrupt threshold
+ * @param[in] upper_threshold           Upper interrupt threshold
  *
  * @return              0 on success
  * @return              -1 on error
  */
 int isl29125_init(isl29125_t *dev, i2c_t i2c, gpio_t gpio,
                   isl29125_mode_t mode, isl29125_range_t range,
-                  isl29125_resolution_t resolution);
+                  isl29125_resolution_t resolution,
+                  isl29125_interrupt_status_t interrupt_status,
+                  isl29125_interrupt_persist_t interrupt_persist,
+                  isl29125_interrupt_conven_t interrupt_conven,
+                  uint16_t lower_threshold,
+                  uint16_t higher_threshold);
```

Comment:
IMHO this makes init of the device rather complex, specifically if interupt is not needed I still have to provide some values for all params. Further, this introduces an API change that breaks existing code using this driver.

I would rather leave the `isl29125_init` function as is and provide an additional call such as `init_irq` with all those extra params.